pkg/utils/namer/interfaces.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namer
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"strings"
+
+	"k8s.io/api/networking/v1beta1"
+	"k8s.io/klog"
+)
+
+const (
+	LegacyNamingScheme = Scheme("v1")
+	V2NamingScheme     = Scheme("v2")
+	maxLength          = 36
+)
+
+type Scheme string
+
+// IngressFrontendNamer is an interface to name GCE frontend resources.
+type IngressFrontendNamer interface {
+	ForwardingRule(protocol NamerProtocol) string
+	TargetProxy(protocol NamerProtocol) string
+	UrlMap() string
+	SSLCertName(secretHash string) string
+	IsCertUsedForLB(resourceName string) bool
+	IsLegacySSLCert(resourceName string) bool
+	GetBENamer() *Namer
+	GetLbName() string
```

Comment:
GetLbName need better handling:

Consider having a different type for lbName. This will make the logic less error prune. (e.g. pass something other than lbname to downstream consumer)

```
type LoadBalancerName string


GetLbName() LoadBalancerName
```

Reply:
GetLbName does not have a lot of significance while creating or gcing resources. This is more for logging. I think its okay to keep it as string.

Reply:
I can get rid of this function if that makes it better.